superset-frontend/src/components/Select/utils.tsx

```diff
@@ -49,22 +49,25 @@ export function getValue(
   return isLabeledValue(option) ? option.value : option;
 }
 
+export function equalValues(a: V | LabeledValue, b: V | LabeledValue) {
+  const actualA = isObject(a) && 'value' in a ? a.value : a;
+  const actualB = isObject(b) && 'value' in b ? b.value : b;
+  // When comparing the values we use the equality
+  // operator to automatically convert different types
+  // eslint-disable-next-line eqeqeq
+  return actualA == actualB;
+}
+
 export function getOption(
   value: V,
   options?: V | LabeledValue | (V | LabeledValue)[],
   checkLabel = false,
 ): V | LabeledValue {
   const optionsArray = ensureIsArray(options);
-  // When comparing the values we use the equality
-  // operator to automatically convert different types
   return optionsArray.find(
     x =>
-      // eslint-disable-next-line eqeqeq
-      x == value ||
-      (isObject(x) &&
-        // eslint-disable-next-line eqeqeq
-        (('value' in x && x.value == value) ||
-          (checkLabel && 'label' in x && x.label === value))),
+      equalValues(x, value) ||
+      (checkLabel && isObject(x) && 'label' in x && x.label === value),
```

Comment:
Do you think it makes sense to extend `equalValues` to accept "key" argument, so you could call `equalValues(x, checkLabel, "label")` here?

Reply:
Good call. Done.